### PR TITLE
[hidapi] Add support linux and osx

### LINKED ISSUE #15644: [hidapi] Linux/macOS support
`hidapi` is available for Linux, macOS and Windows, but the vcpkg port only supports the latter.

### LINKED ISSUE #16918: [hidapi] Support for x64-linux
I am trying to install the hidapi port on Linux Debian 10 for some project.

The output of ```uname -a``` is:
```
Linux debian 4.19.0-14-amd64 #1 SMP Debian 4.19.171-2 (2021-01-30) x86_64 GNU/Linux
```
When I try to run:
```
./vcpkg install --overlay-ports=/home/nick-test/<project_specific>/custom_ports --overlay-triplets=/home/nick-test/<project_specific>/vcpkg/custom_triplets --triplet x64-linux hidapi
```
I get the following output:
```
The following packages will be built and installed:
    hidapi[core]:x64-linux -> 0.10.1
...
 Target 'linux' not supported by hidapi!
Please ensure you're using the latest portfiles with `./vcpkg update`, then
submit an issue at https://github.com/Microsoft/vcpkg/issues including:
  Package: hidapi:x64-linux
  Vcpkg version: 2021-01-13-unknownhash

```

Could the hidapi:x64-linux combination be added to the hidapi port?

### PR DESCRIPTION
**Pull request by NancyLi1013**
**Describe the pull request**

- What does your PR fix? Fixes #16918, Fixes #15644

Note: No feature needs to test.

### CODE REVIEW

**Comment by NancyLi1013** [comment]
The failures on osx pipeline:

```
autoreconf: export WARNINGS=
autoreconf: Entering directory '.'
autoreconf: configure.ac: not using Gettext
autoreconf: running: aclocal --force -I m4
configure.ac:16: error: AC_CONFIG_MACRO_DIR can only be used once
./lib/autoconf/general.m4:1970: AC_CONFIG_MACRO_DIR is expanded from...
configure.ac:16: the top level
autom4te: error: /usr/local/opt/m4/bin/m4 failed with exit status: 1
aclocal: error: /usr/local/Cellar/autoconf/2.71/bin/autom4te failed with exit status: 1
autoreconf: error: aclocal failed with exit status: 1
```

**Review by JackBoosY — APPROVED** [review]

**Comment by JackBoosY** [comment]
```
CMake Error at ports/popsift/portfile.cmake:9 (include):
  include could not find load file:

    D:/installed/x64-windows/share/vcpkg_find_cuda/vcpkg_find_cuda.cmake
Call Stack (most recent call first):
  scripts/ports.cmake:142 (include)
```
Is this related to your changes?

**Comment by autoantwort** [comment]
> The failures on osx pipeline:
> 
> ```
> autoreconf: export WARNINGS=
> autoreconf: Entering directory '.'
> autoreconf: configure.ac: not using Gettext
> autoreconf: running: aclocal --force -I m4
> configure.ac:16: error: AC_CONFIG_MACRO_DIR can only be used once
> ./lib/autoconf/general.m4:1970: AC_CONFIG_MACRO_DIR is expanded from...
> configure.ac:16: the top level
> autom4te: error: /usr/local/opt/m4/bin/m4 failed with exit status: 1
> aclocal: error: /usr/local/Cellar/autoconf/2.71/bin/autom4te failed with exit status: 1
> autoreconf: error: aclocal failed with exit status: 1
> ```

I am now getting the same error in the pipeline of my PRs for this port?

File: ports/hidapi/portfile.cmake [hunk 1]
```diff
@@ -1,11 +1,12 @@
-vcpkg_fail_port_install(ON_ARCH "arm" "arm64" ON_TARGET "linux" "osx" "uwp")
+vcpkg_fail_port_install(ON_ARCH "arm" "arm64" ON_TARGET "uwp")
 
 vcpkg_from_github(
     OUT_SOURCE_PATH SOURCE_PATH
     REPO libusb/hidapi
     REF hidapi-0.10.1
     SHA512 0479706c631775483378070ff7170542725678eabc202a5bd07436c951fd766e01743417999ac3fb2b5436c865f6ace2cfced1f210fa3a3e88c19ceb3bbe0534
     HEAD_REF master
+    patch remove-duplicate-AC_CONFIG_MACRO_DIR.patch
```
Comment (autoantwort) [inline ports/hidapi/portfile.cmake:9]:
```suggestion
    PATCHES remove-duplicate-AC_CONFIG_MACRO_DIR.patch
```
Maybe it is possible that the functions throws an error is you pass unknown options to it. And I wounder a lot why the pipeline was successful

Comment (JackBoosY) [reply ports/hidapi/portfile.cmake:9]: Sorry for my wrong review.

Comment (NancyLi1013) [reply ports/hidapi/portfile.cmake:9]: Thanks for pointing this out. @autoantwort Sorry for my mistake.

**Pull request merged**